src/goto-instrument/cover_instrument_other.cpp

```diff
@@ -75,10 +75,19 @@ void cover_instrument_end_of_function(
   const irep_idt &function_id,
   goto_programt &goto_program)
 {
-  auto if_it = goto_program.instructions.end();
-  while(!if_it->is_function_call())
-    if_it--;
-  if_it++;
+  const auto last_function_call = std::find_if(
+    goto_program.instructions.rbegin(),
+    goto_program.instructions.rend(),
+    [](const goto_programt::instructiont &instruction) {
+      return instruction.is_function_call();
+    });
+  INVARIANT(
+    last_function_call != goto_program.instructions.rend(),
+    "Goto program should have at least one function call");
+  INVARIANT(
+    last_function_call != goto_program.instructions.rbegin(),
+    "Goto program shouldn't end with a function call");
+  const auto if_it = std::next(last_function_call.base());
```

Comment:
Should this be `std::prev`?

Reply:
No, because I called `.base()` to convert from reverse iterator to forward iterator.

Reply:
Yes, but `.base()` gives an iterator one to the right to the element that dereferencing the reverse iterator would yield. Thus to get a forward iterator to the same position we need to go back again via `std::prev`